### DISCUSSION TITLE
Error during build: You must supply options.input to rollup

### DISCUSSION THREAD

**Discussion by Augustin-Grenne** [ElMassimo/vite_ruby#459] [Q&A] [answered]
I followed these very neat install instructions: https://vite-ruby.netlify.app/guide/rails.html but unfortunately I'm running into an error when starting a dev server. It tells me to run `bin/vite build --clear --mode=development` but doing so results in the same error: 
```
Removed vite cache and output dirs:
	/Users/augustin/code/bpart/public/vite-dev
	/Users/augustin/code/bpart/public/vite-ssr
	/Users/augustin/code/bpart/tmp/cache/vite
	/Users/augustin/code/bpart/node_modules/.vite
Building with Vite ⚡️
vite v5.2.11 building for development...
✓ 0 modules transformed.
x Build failed in 2ms
error during build:
Error: You must supply options.input to rollup
    at Graph.generateModuleGraph (file:///Users/augustin/code/bpart/node_modules/rollup/dist/es/shared/node-entry.js:20000:19)
    at async Graph.build (file:///Users/augustin/code/bpart/node_modules/rollup/dist/es/shared/node-entry.js:19966:9)
    at async file:///Users/augustin/code/bpart/node_modules/rollup/dist/es/shared/node-entry.js:20615:13
    at async catchUnfinishedHookActions (file:///Users/augustin/code/bpart/node_modules/rollup/dist/es/shared/node-entry.js:20123:16)
    at async rollupInternal (file:///Users/augustin/code/bpart/node_modules/rollup/dist/es/shared/node-entry.js:20610:5)
    at async build (file:///Users/augustin/code/bpart/node_modules/vite/dist/node/chunks/dep-cNe07EU9.js:67475:18)
    at async CAC.<anonymous> (file:///Users/augustin/code/bpart/node_modules/vite/dist/node/cli.js:842:9)

pid 98761 exit 1
Build with Vite failed! ❌
```

Does anyone know which `options.input` is meant and what's expected? I find nothing about this in the documentation or troubleshooting sections. Googling the issue only led me to some people with seemingly very different set ups and/or the issue was related to some plugins we're not using.

Some quick context, I'm working on a rather large and complex, monolithic Rails app. Which in the past I've painstakingly moved from using Sprockets to Webpacker and from Webpacker to JS bundling with ESbuild. While our current implementation works, it always felt a little sub-optimal. We're missing out on the easy live reload, HMR, etc. I've took 2 stabs at moving to using vite in the past but encountered different issues and had to move on, to do other stuff. I hope, with a little help of this community, the third time's going to be the charm.

**Comment by ElMassimo** [comment, marked as answer]
Hi Augustin!

You shouldn't need to provide `input`, as that's [inferred by `vite-plugin-ruby`](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/src/index.ts#L48) from your [Entrypoints](https://vite-ruby.netlify.app/guide/development.html#entrypoints-%E2%A4%B5%EF%B8%8F).

Try [debugging](https://vite-ruby.netlify.app/guide/debugging.html#getting-more-output-from-vite-%F0%9F%93%9C) the dev server, to verify that your configuration is correct, and that you have entrypoints to be bundled.

**Comment by Augustin-Grenne** [comment]
Thanks for the quick response. I thought as much about the input, but still was confused. Turns out indeed I did not correctly update the `sourceCodeDir` after moving assets about. Sorry about that. 
Follow up question though, what's your opinion on a good file structure? Currently we have `app/assets/stylesheets` with some scss 'entryfiles' importing a bunch of partials and `app/javascript` which also contains some 'entryfiles' as well as some lose scripts and AlpineJS components. I'm leaning towards combining that in `app/frontend` but I've seen set up's with `frontend` in the root or keeping the split dirs with defining additional entrypoints. 
Anyhow thanks for your help. I'm off redefining some import paths in my asset files.

**Reply by ElMassimo** [reply]
> I'm off redefining some import paths in my asset files.

To simplify the migration, consider using [aliases](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89), you can [add your own as a temporary measure](https://vitejs.dev/config/shared-options.html#resolve-alias), such as `@assets`.

> What's your opinion on a good file structure?

I tend to favor `app/frontend`, as it's signals that you'll have JS and CSS there, and that it's part of your Rails app.
